examples/kubernetes/addons/etcd-operator/README.md

```diff
@@ -0,0 +1,90 @@
+Kubernetes etcd-operator integration (beta)
+===========================================
+
+This directory contains all the necessary files to deploy etcd-operator with
+Cilium in your development cluster.
+
+Create etcd certificates
+------------------------
+
+The first step you need to do is
+
+```
+tls/certs/gen-cert.sh <cluster domain>
+```
+where `<cluster domain>` is the domain the cluster set up in kube-dns.
+
+You can find it by checking the config map of core-dns by running
+```
+kubectl get ConfigMap --namespace kube-system coredns -o yaml | grep kubernetes
+```
+
+or by checking the kube-dns deployment and grepping for 'domain'
+```
+kubectl get Deployment --namespace kube-system kube-dns -o yaml | grep domain
+```
+
+For reference, the cluster domain used in Kubernetes clusters by default is `'cluster.local'`
+
+Deploy generated certificates
+-----------------------------
+
+The next step is to deploy the certificates generated on the previous step
+in the Kubernetes cluster, this can be achieved by running.
+
+```
+tls/deploy-certs.sh
+```
+
+Deploy kube-dns and make sure it contains the required label
+------------------------------------------------------------
+
+Before running this step, make sure you have kube-dns running and contains
+the label `io.cilium.fixed-identity=kube-dns`.
+
+```
+kubectl label -n kube-system pod $(kubectl get pods -n kube-system | grep kube-dns | cut -f 1 -d " ") io.cilium.fixed-identity=kube-dns
```

Comment:
Use JSON parsing instead of grep / cut; this has proven brittle for us in the past due to the K8s CLI formatting changes.

Reply:
Will do